src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -279,6 +280,88 @@ class MergeIntoScalaSuite extends MergeIntoSuiteBase {
     }
   }
 
+  test("findTouchedFiles - no files match target predicate") {
+    val jobCount = new AtomicInteger(0)
+    val listener = new SparkListener {
+      override def onJobStart(jobStart: SparkListenerJobStart): Unit = {
+        // Spark will always log a job start/end event even when the job does not launch any task.
+        if (jobStart.stageInfos.exists(_.numTasks > 0)) {
+          jobCount.incrementAndGet()
```

Comment:
this listener implementation can be deduplicated. across the tests. You can define a method called 
```
def withJobCountListener(thunk: => Unit): Int = {
   // create listener
   // run thunk
   // return job count from listener
}
```

Reply:
@tdas should I put it in something similar to Spark's `SQLTestUtilsBase` or keep it local to this test suite?

Reply:
Its a very specific functionality to count the jobs, and I dont imagine this be useful outside these two tests. So just put it locally before the tests.